GDJS/tests/tests/objecttools.js

```diff
@@ -127,4 +92,62 @@ describe('gdjs.evtTools.object', function () {
       )
     ).to.be(0);
   });
+
+  const getInstancesIds = (instances) =>
+    instances.map((instance) => instance && instance.id);
+
+  it('can create and pick an object when an instance is already picked', function () {
+    const runtimeGame = gdjs.getPixiRuntimeGame();
+    const runtimeScene = new gdjs.TestRuntimeScene(runtimeGame);
+
+    runtimeScene.registerObjectWithName('MyObjectA');
+    const objectA1 = runtimeScene.createObject('MyObjectA');
+    const objectA2 = runtimeScene.createObject('MyObjectA');
+
+    // An instance is already picked.
+    const pickedObjectList = Hashtable.newFrom({
+      MyObjectA: [objectA1],
+    });
+
+    const newObjectA = gdjs.evtTools.object.createObjectOnScene(
+      runtimeScene,
+      pickedObjectList,
+      0,
+      0,
+      ''
+    );
+
+    // The new instance is added to the picked instances.
+    expect(getInstancesIds(pickedObjectList.get('MyObjectA'))).to.eql(
+      getInstancesIds([objectA1, newObjectA])
+    );
+  });
+
+  // TODO The created instance should be picked when the object is free.
+  it('can create an object and keep the object free from picking', function () {
+    const runtimeGame = gdjs.getPixiRuntimeGame();
+    const runtimeScene = new gdjs.TestRuntimeScene(runtimeGame);
+
+    runtimeScene.registerObjectWithName('MyObjectA');
+    const objectA1 = runtimeScene.createObject('MyObjectA');
+    const objectA2 = runtimeScene.createObject('MyObjectA');
+
+    // No instance is picked (as they are all in the list).
```

Comment:
I would avoid this comment, objects are really picked - it's just that they are ALL picked.